docs/specs/user-profile/design.md

```diff
@@ -172,15 +183,18 @@ The pointer MUST NOT conflict with the pointer of any standard attributes,
 so the developer CANNOT define a custom attribute with pointer `/email`.
 
 Once a custom attribute is defined, it CANNOT be removed.
-The pointer and the type CANNOT be changed.
+The `id` and the `type` CANNOT be changed.
 
 > If custom attribute were allowed to be removed, or its type were allowed to be changed, the developer can remove it and define
-> a new custom attribute with the same pointer but a incompatible type.
+> a new custom attribute with the same ID but a incompatible type.
 > This will result in a situation that is very complicated to handle.
 > Given that it is extremely easy to define custom attribute with just a few clicks,
 > a few clicks should never lead to such a complicated scenario.
 
-Other than `pointer` and `type`, the developer is allowed to freely change other supplementary configuration of a particular type of custom attribute.
+The `pointer` can be changed as long as the developer is aware of the consequence of the rename.
+They have to be prepared for receiving custom attributes shown in the new pointer.
+
+The developer is allowed to freely change other supplementary configuration of a particular type of custom attribute.
```

Comment:
How do we handle the case if the custom attribute had a negative value, and developer add `minimum: 0.0` to the config?

Reply:
During read, we do not perform validation so the read will succeed.
The input control will be filled with the invalid value.
When the value is written with the value just read, the write will fail.
So error will delay until the custom attribute is written.

Reply:
If the field `a` fall into this situation, then user change field `b` value in the settings (given that the ui only allow update single field). Can user save the field `b` successfully?

Reply:
Yes, correct. Though in the portal, the admin user will likely see all fields at the same time, they need to correct `a` in order to save everything successfully.